dev/tools/localization/bin/gen_localizations.dart

```diff
@@ -562,17 +579,35 @@ void main(List<String> rawArgs) {
   );
 
   try {
+    validateLocalizations(widgetsLocaleToResources, widgetsLocaleToResourceAttributes, removeUndefined: options.removeUndefined);
     validateLocalizations(materialLocaleToResources, materialLocaleToResourceAttributes, removeUndefined: options.removeUndefined);
     validateLocalizations(cupertinoLocaleToResources, cupertinoLocaleToResourceAttributes, removeUndefined: options.removeUndefined);
   } on ValidationError catch (exception) {
     exitWithError('$exception');
   }
-
   if (options.removeUndefined) {
+    removeUndefinedLocalizations(materialLocaleToResources);
```

Comment:
Should this be widgetsLocaleToResources? Maybe add a test that would have failed for this case